domenetjenester/uttak/src/main/java/no/nav/foreldrepenger/domene/uttak/svp/OppholdTjeneste.java

```diff
@@ -0,0 +1,118 @@
+import no.nav.foreldrepenger.domene.arbeidsforhold.InntektsmeldingTjeneste;
+import no.nav.foreldrepenger.domene.typer.InternArbeidsforholdRef;
+import no.nav.foreldrepenger.domene.uttak.input.SvangerskapspengerGrunnlag;
+import no.nav.svangerskapspenger.domene.felles.AktivitetType;
+import no.nav.svangerskapspenger.domene.felles.Arbeidsforhold;
+import no.nav.svangerskapspenger.domene.søknad.Opphold;
+import no.nav.svangerskapspenger.tjeneste.fastsettuttak.SvpOppholdÅrsak;
+
+@ApplicationScoped
+public class OppholdTjeneste {
+    private InntektsmeldingTjeneste inntektsmeldingTjeneste;
+
+    @Inject
+    OppholdTjeneste(InntektsmeldingTjeneste inntektsmeldingTjeneste) {
+        this.inntektsmeldingTjeneste = inntektsmeldingTjeneste;
+    }
+    public OppholdTjeneste() {
+        //CDI
+    }
+    public Map<Arbeidsforhold, List<Opphold>> finnOppholdFraTilretteleggingOgInntektsmelding(BehandlingReferanse behandlingRef, SvangerskapspengerGrunnlag svpGrunnlag) {
+        Map<Arbeidsforhold, List<Opphold>> oppholdForAlleArbeidsforholdMap = new HashMap<>();
+        svpGrunnlag.getGrunnlagEntitet()
+            .map(SvpGrunnlagEntitet::hentTilretteleggingerSomSkalBrukes)
+            .orElse(Collections.emptyList()).forEach(tilrettelegging -> {
+            var aktivitetType = mapTilAktivitetType(tilrettelegging.getArbeidType());
+            var arbeidsforhold =  lagArbeidsforhold(tilrettelegging.getArbeidsgiver(), tilrettelegging.getInternArbeidsforholdRef(), aktivitetType);
+            //henter ferie- eller sykepenger-opphold registrert av saksbehandler
+            var oppholdListeFraTIlr = hentOppholdListeFraTilrettelegging(tilrettelegging);
+            List<Opphold> alleOppholdForArbeidsforhold = new ArrayList<>(oppholdListeFraTIlr);
+
+            //henter ferier fra inntektsmelding meldt av arbeidsgiver
+            tilrettelegging.getArbeidsgiver().ifPresent( arbeidsgiver -> {
+                var oppholdListeArbeidsforholdFraInntektsmelding = finnOppholdFraIMForArbeidsgiver(behandlingRef, arbeidsgiver, tilrettelegging.getInternArbeidsforholdRef().orElse(InternArbeidsforholdRef.nullRef()));
+                alleOppholdForArbeidsforhold.addAll(oppholdListeArbeidsforholdFraInntektsmelding);
+            });
+
+            oppholdForAlleArbeidsforholdMap.put(arbeidsforhold, alleOppholdForArbeidsforhold);
+
+        });
+        return oppholdForAlleArbeidsforholdMap;
+    }
+
+    private List<Opphold> hentOppholdListeFraTilrettelegging(SvpTilretteleggingEntitet tilrettelegging) {
+        List<Opphold> oppholdListeFraSaksbehandler = new ArrayList<>();
+        tilrettelegging.getAvklarteOpphold().forEach( avklartOpphold -> oppholdListeFraSaksbehandler.addAll(Opphold.opprett(avklartOpphold.getFom(), avklartOpphold.getTom(), mapOppholdÅrsak(avklartOpphold.getOppholdÅrsak()))));
+        return oppholdListeFraSaksbehandler;
+    }
+
+    private Arbeidsforhold lagArbeidsforhold(Optional<Arbeidsgiver> arbeidsforhold,
```

Comment:
I stedet for å ha en egen metode for å lage arbeidsforhold her, kanskje heller gjøre den i RegelmodellSøknaderMapper static og kalle på den? Så vet vi at alle feltene settes likt

Reply:
fikset